### PR TITLE
python310Packages.opentelemetry: unified the source code

### PR DESCRIPTION
**Pull request by natsukium**
###### Description of changes

It was pointed out on [matrix](https://matrix.to/#/!VjfUzaKsXokUdnQcvP:nixos.org/$9DnbZKGlZacc8xm4umLfG1rsOYnd1VW8f28E6J2sEWM?via=nixos.org&via=matrix.org&via=nixos.dev) that each module under `open-telemetry/opentelemetry-python` is packaged separately and, therefore, fragile to version updates.

In this PR, I unified the `version`, `src`, and `meta` using the core package, `opentelemetry-api`, as the source.

I also added myself as a maintainer to make it easier to notice changes related to opentelemetry-api.

I hope we can package them in a better way.

Also, I will work on `opentelemetry-python-contrib` after this PR and #245028 are done.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/python-modules/opentelemetry-api/default.nix [hunk 1]
```diff
@@ -17,13 +17,15 @@ let
     version = "1.18.0";
     disabled = pythonOlder "3.7";
 
+    # to avoid breakage, every package in opentelemetry-python must inherit this version, src, and meta
```
Comment (RaitoBezarius) [inline pkgs/development/python-modules/opentelemetry-api/default.nix:20]: It's not heriting src here, is it?

File: pkgs/development/python-modules/opentelemetry-exporter-otlp-proto-common/default.nix [hunk 2]
```diff
@@ -1,26 +1,20 @@
...
+, opentelemetry-api
 , opentelemetry-proto
 , opentelemetry-sdk
 , opentelemetry-test-utils
 , pytestCheckHook
 }:
 
-buildPythonPackage rec {
+buildPythonPackage {
+  inherit (opentelemetry-api) version src;
```
Comment (RaitoBezarius) [inline pkgs/development/python-modules/opentelemetry-exporter-otlp-proto-common/default.nix:13]: Ah I see!

**Comment by RaitoBezarius** [comment]
Can we ping the DeShaw team too here?

**Review by SuperSandro2000 — APPROVED** [review]

**Comment by natsukium** [comment]
> Can we ping the DeShaw team too here?

Ofcource, ofborg should already ping them. 
@limeytexan san, I'll ping you just in case.

**Comment by happysalada** [comment]
would you mind adding grpc and aiohttp-client as well, I've just merged https://github.com/NixOS/nixpkgs/pull/245028

**Review by de11n — APPROVED** [review]
Thanks for doing this. I was unsure if this was the best approach when adding these packages originally, but I'm fine with doing it this way. FYI: I'm the original author of these packages and on the deshaw team, but the members are out of date (see #245907).

**Comment by natsukium** [comment]
I did the same for `opentelemetry-python-contrib` and added commits.
I can drop them again if another PR is preferred.

**Comment by happysalada** [comment]
This looks good to me, waiting one day in case anyone else wants to comment.

**Review by adisbladis — APPROVED** [review]

**Pull request merged**